src/libs/runtime-utils.js

```diff
@@ -222,33 +223,37 @@ export const machineCost = machineType => {
   return _.find(knownMachineType => knownMachineType.name === machineType, machineTypes).price
 }
 
-export const getCurrentApp = _.flow(trimAppsOldestFirst, _.last)
+export const getCurrentAppForType = appType => _.flow(trimAppsOldestFirst, _.remove(_ => _.appType !== appType), _.last)
 
-export const currentAppIncludingDeleting = _.flow(_.sortBy('auditInfo.createdDate'), _.last)
+// DEPRECATED: this assumes that Galaxy is the only app that could be running.
+// Use getCurrentAppForType instead.
+export const getCurrentApp = getCurrentAppForType(tools.galaxy.appType)
```

Comment:
Is it worth adding a console log along the lines of "you used a deprecated function" here?